httpx/_multipart.py

```diff
@@ -20,6 +20,18 @@
 )
 
 
+def get_multipart_boundary_from_content_type(
+    content_type: typing.Optional[str],
+) -> typing.Optional[bytes]:
+    if not content_type or not content_type.startswith("multipart/form-data"):
+        return None
+    if ";" in content_type:
+        for section in content_type.split(";"):
+            if section.strip().startswith("boundary="):
+                return section.strip().split("boundary=")[-1].encode("latin-1")
```

Comment:
Is there any possibility that `boundary` could be quoted, with either simple or double quotes?

Reply:
[RFC 2046](https://www.rfc-editor.org/rfc/rfc2046#section-5.1.1) has examples where boundary is contained in quotes...

Reply:
Great catch! Not sure how you find these things but thank you

Reply:
After reviewing the grammar for boundary ([RFC 2046 Appendix A](https://www.rfc-editor.org/rfc/rfc2046#page-43), referencing [RFC 2045 §5.1](https://www.rfc-editor.org/rfc/rfc2045#section-5.1), referencing [RFC 822 §3.3](https://www.rfc-editor.org/rfc/rfc822#section-3.3) which has been replaced by [RFC 5322 §3.2.4](https://www.rfc-editor.org/rfc/rfc5322#section-3.2.4)):

1. The only quoting character that seems to be valid is the double quote `"` (`'` is not valid for that usage);
2. However, when the double quote is in use, any backslash `\` will be part of a "quoted pair", meaning that it is semantically equivalent to the next character (`\\` => `\`, `\"` => `"`);
3. However, there is a limitation on allowed characters in a boundary - none of the characters that would require a quoted pair escape are allowed in a boundary;

In other words:
 * We need to handle only `"` for quoting;
 * We also need to decide what to do about quoted pairs:
   * Either we write a simpler implementation that does not handle them, as all legal boundaries can be written without using quoted pairs;
   * Or we go the extra mile of supporting quoted pairs;
 * What do we want to do about the fact that not all ascii characters are allowed as part of a boundary?

Reply:
I think we can go with the simpler implementation. I also don't think we need to handle non-valid characters, that'd be too much introspection